server/src/upload.php

```diff
@@ -20,21 +20,22 @@
     exit();
 }
 
+$mimetype = isset($screenshot['tmp_name']) ? mime_content_type($screenshot['tmp_name']) : "error";
 if (
     // Make sure it is a successful single file upload
     !isset($screenshot['error']) ||
     is_array($screenshot['error']) ||
     $screenshot['error'] !== UPLOAD_ERR_OK ||
     // Verify size and mime type
     $screenshot['size'] > $config['maxScreenshotSize'] ||
-    mime_content_type($screenshot['tmp_name']) !== $config['screenshotMimeType']
+    !in_array( $mimetype, $config['screenshotMimeTypes'], true)
 ) {
     http_response_code(422);
     exit();
 }
 
 // Generate screenshot path
-do {} while (file_exists($filename = bin2hex(random_bytes(12)) . '.png'));
+do {} while (file_exists($filename = bin2hex(random_bytes(12)) . '.' . ($mimetype == "image/png" ? "png" : "jpg" )));
```

Comment:
If someone adds other mimetypes besides these 2, I guess we can't really have this assumption of using `.jpg` here.